Sprint 13

UtinsStory opened the pull request:
Работа по спринту 13

UtinsStory committed:
Создан WebViewPresenter. Часть логики перенесена из WebViewController в WebViewPresenter

UtinsStory committed:
Создан AuthHelper. Часть логики перенесена из WebViewPresenter в AuthHelper

UtinsStory committed:
Рефакторинг ProfileViewController

UtinsStory committed:
Исправлена ошибка в рефакториге ProfileVC

UtinsStory committed:
Рефакторинг ImagesListViewController

UtinsStory committed:
Добавлены unit-тесты для профиля

UtinsStory committed:
Добавлены unit-тесты для ленты изображений

UtinsStory committed:
Добавлен ui-тест авторизации

UtinsStory committed:
Попытка написания UI-тестов

UtinsStory committed:
Вторая попытка написания тестов

UtinsStory committed:
Третья попытка написания тестов

UtinsStory committed:
Сдача задачи спринта 13 на ревью
//в UI-тестах расставлены слипы с большим значением из-за проблем с интернет-соединением на моей стороне

andrew-kuzn-yndx commented on Image Feed/Auth/AuthViewController.swift:
```diff
@@ -43,7 +50,7 @@ final class AuthViewController: UIViewController,
     //MARK: IB Actions
     
     @IBAction func didTapLoginButton() {
```
Не забывай, что IBAction **всегда** должны быть приватными 
Ну и тудушка не лишняя пока)

andrew-kuzn-yndx commented on Image Feed/Helpers/AuthHelper.swift:
```diff
@@ -0,0 +1,56 @@
+    
+    func code(from url: URL) -> String? {
+        if let urlComponents = URLComponents(string: url.absoluteString),
+           urlComponents.path == "/oauth/authorize/native",
```
Давай строки path тоже поместим в файл Constants, который у тебя есть) используй его по полной

andrew-kuzn-yndx commented on Image Feed/ImagesList/ImagesListViewController.swift:
```diff
@@ -25,17 +26,13 @@ final class ImagesListViewController: UIViewController {
         super.viewDidLoad()
         
+//        presenter = ImagesListPresenter()
+//        presenter?.view = self
```
Старайся не оставлять такое, правилом хорошего тона является чистить код от лишнего перед сдачей работы. В основную ветку проекта должен попадать только чистый код без лишних принтов и закомментированных блоков)

andrew-kuzn-yndx commented on Image Feed/ImagesList/ImagesListViewController.swift:
```diff
@@ -66,52 +72,42 @@ final class ImagesListViewController: UIViewController {
+        
         cell.cellImage.kf.indicatorType = .activity
         cell.cellImage.kf.setImage(with: imageUrl, placeholder: UIImage(named: "feed_placeholder"))
         cell.dateLabel.text = self.dateFormatter.string(for: photo.createdAt)
```
Старайся не работать со свойствами ячейки и вообще класса вот так напрямую из другого класса. 
По хорошему - это сделать в ячейке публичную функцию, например configCell с аргументами на вход, и уже в ней менять нужные свойства. 
Свойства класса нужно всегда стараться закрывать извне, и работать с ними только изнутри

andrew-kuzn-yndx commented on Image Feed/Profile/ProfileViewController.swift:
```diff
@@ -49,6 +59,11 @@ final class ProfileViewController: UIViewController {
     
     //MARK: - METHODS
```
Молодец, что указал здесь MARK
 Однако, было бы здорово делать этот систематически - всегда и везде использовать это разделение, организовать это во всём проекте
Вот тут больше информации по теме: https://habr.com/ru/articles/744470/:

UtinsStory committed:
Проект исправлен для запуска в тестовой среде ревьюера
//предполагаю, что проект не запускался, так как в новой версии xcode папки с тестами создались как folders. переделал в поддерживаемые xcode 15.4 папки groups//
